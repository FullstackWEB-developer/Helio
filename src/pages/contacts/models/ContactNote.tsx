export interface ContactNote {
    id?: number;
    noteText: string;
    createdOn?: Date;
    createdBy?: string;
}