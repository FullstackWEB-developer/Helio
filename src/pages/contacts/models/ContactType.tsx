export enum ContactType {
    Individual = 'Individual',
    Company = 'Company'
}