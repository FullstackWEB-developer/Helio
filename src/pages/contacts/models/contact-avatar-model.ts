import {Icon} from '@components/svg-icon/icon';
export interface ContactAvatarModel {
    initials?: string;
    className?: string; 
    isCompany?: boolean;
    iconType?: Icon;
    iconClassName?: string;
    iconFillClass?: string;
}