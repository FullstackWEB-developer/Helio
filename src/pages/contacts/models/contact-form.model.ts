import {Option} from "@components/option/option";

export interface ContactFormModel {
    id?: string;
    firstName?: string;
    lastName?: string;
    companyName?: string;
    category: number | Option;
    jobTitle?: string;
    department?: string;
    email?: string;
    workMainPhone?: string;
    workMainExtension?: string;
    workDirectPhone?: string;
    mobile?: string;
    fax?: string;
    website?: string;
    primaryAddressLine?: string;
    primaryApt?: string;
    primaryCity?: string;
    primaryState?: string | Option;
    primaryZipCode?: string;
    shippingAddressLine?: string;
    shippingApt?: string;
    shippingCity?: string;
    shippingState?: string | Option;
    shippingZipCode?: string;
    billingAddressLine?: string;
    billingApt?: string;
    billingCity?: string;
    billingState?: string | Option;
    billingZipCode?: string;
    isStarred?: boolean;
    relatedId?: string;
}