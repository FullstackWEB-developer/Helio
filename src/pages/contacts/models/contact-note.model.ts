export interface ContactNote {
    id?: string;
    noteText: string;
    createdOn: Date;
    createdByName: string;
    contactId: string;
}
export interface AddContactNoteProps {
    contactId: string;
    contactNoteDto: ContactNote
}