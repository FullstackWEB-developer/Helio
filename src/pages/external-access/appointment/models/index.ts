export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
