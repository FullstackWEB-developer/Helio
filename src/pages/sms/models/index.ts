export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
