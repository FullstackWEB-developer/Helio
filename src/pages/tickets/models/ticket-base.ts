import {ChannelTypes} from "@shared/models/ticket-channel";
import {TicketType} from '@shared/models';
import {TicketStatuses} from '@pages/tickets/models/ticket.status.enum';

export interface TicketBase {
    id: string;
    ticketNumber: number;
    status?: TicketStatuses;
    type: TicketType;
    priority?: number
    channel: ChannelTypes;
    subject: string;
    dueDate?: Date;
    createdOn?: Date;
    closedOn?: Date;
    isOverdue?: boolean;
    patientId?: number;
    contactId: string;
    reason: string;
    assignee?: string;
    createdForName: string;
    createdBy: string;
}
