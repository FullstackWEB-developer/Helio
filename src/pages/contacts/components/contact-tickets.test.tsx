import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import ContactTickets from './contact-tickets';
import Api from '@shared/services/api';
import {TicketBase} from '@pages/tickets/models/ticket-base';
import {ChannelTypes, TicketType} from '@shared/models';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("ContactTickets tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            },
            auth: {
                name: "Test"
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: [{
                label: "Test",
                parentValue: "Test"
            },{
                label: "Test",
                parentValue: "Test"
            }]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        ccpState: {
            chatCounter: 1,
            voiceCounter: 2
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders ContactTickets isDanger true correctly", async () => {
        jest.spyOn(Api, 'get').mockReturnValue({
            data: {
                pageSize: 1,
                page: 1,
                totalCount: 10,
                totalPages: 10,
                results: [
                    {
                        contactId: '123',
                        id:'123',
                        channel: ChannelTypes.Chat,
                        type: TicketType.BusinessOffice,
                        createdBy: 'burak',
                        createdForName: 'burak',
                        reason: 'reason',
                        subject: 'subject',
                        ticketNumber: 3
                    }
                ] as TicketBase[]
            }
        } as any);
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <ContactTickets contactId='123'/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
