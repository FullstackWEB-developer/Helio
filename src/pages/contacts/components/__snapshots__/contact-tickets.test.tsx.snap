// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTickets tests renders ContactTickets isDanger true correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between py-3 border-b"
  >
    <div
      class="flex items-center body2"
    >
      <div
        class=""
      >
        <svg
          class="pl-1 cursor-pointer icon-large "
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
          />
          <path
            class="active-item-icon"
            d="M19,13H13v6H11V13H5V11h6V5h2v6h6Z"
          />
        </svg>
      </div>
      <span
        class="pl-2 cursor-pointer contact-accent-color hover:underline"
      >
        contacts.contact_details.create_ticket
      </span>
    </div>
  </div>
  <div
    class="flex flex-row items-center content-center w-full h-12 px-6 py-4 bg-gray-100 auto-cols-min body2-medium"
  >
    <div
      class="uppercase flex items-center w-24 "
    >
      tickets.channel
    </div>
    <div
      class="uppercase flex items-center w-2/12 cursor-pointer"
    >
      tickets.id
    </div>
    <div
      class="uppercase flex items-center w-3/12 "
    >
      tickets.subject
    </div>
    <div
      class="uppercase flex items-center w-3/12 cursor-pointer"
    >
      tickets.due_in
    </div>
    <div
      class="uppercase flex items-center w-2/12 cursor-pointer"
    >
      tickets.status
    </div>
  </div>
  <div
    class="flex flex-col space-y-4 items-center h-full w-full justify-center pt-4"
  >
    <div>
      <div
        class=""
      >
        <svg
          class="spinner icon-large-40 "
          viewBox="0 0 40 40"
        >
          <circle
            class="path"
            cx="20"
            cy="20"
            fill="none"
            r="16"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
